src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp

```diff
@@ -685,6 +696,16 @@ namespace Aws
             {
                 ss << ".cn";
             }
+            ss.str();
+
+            Aws::Http::URI uri(ss.str());
```

Comment:
instead of validating the the URI in the constructor, validate it when we make the call like how we currently do it in [AwsClient](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/client/AWSClient.cpp#L262), i.e. in [`AWSHttpResourceClient::GetResourceWithAWSWebServiceResult`](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp#L132C45-L132C102) we should validate there before making any calls